app/controllers/ApplicationController.scala

```diff
@@ -0,0 +1,70 @@
+package controllers
+
+import models.{DataModel, GoogleBook}
+import models.DataModel.formats
+import models.GoogleBook.formats
+import play.api.libs.json.Format.GenericFormat
+import play.api.libs.json.{JsError, JsSuccess, JsValue, Json}
+import play.api.mvc._
+import repositories.DataRepository
+import services.ApplicationService
+
+import javax.inject._
+import scala.concurrent.{ExecutionContext, Future}
+
+
+@Singleton
+class ApplicationController @Inject()(val controllerComponents: ControllerComponents, val datarepository: DataRepository, val service: ApplicationService, implicit val ec: ExecutionContext) extends BaseController {
+
+  def index(): Action[AnyContent] = Action.async { implicit request =>
+    datarepository.index().map {
+      case Right(item: Seq[DataModel]) => Ok {
+        Json.toJson(item)
+      }
+      case Left(error) => Status(error)(Json.toJson("Unable to find any books"))
```

Comment:
I'm not sure if `Status(error)` works - it seems like Status expects to take an integer. I think you could wrap it in a `NoContent("Unable to find any books")` here